packages/grid/_modules_/grid/components/ScrollArea.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import { COL_REORDER_START, COL_REORDER_STOP, SCROLLING } from '../constants/eventsConstants';
+import { ScrollParams, useApiEventHandler } from '../hooks';
+import { ApiRef } from '../models';
+import { classnames } from '../utils';
+import { ApiContext } from './api-context';
+
+interface ScrollAreaProps {
+  scrollDirection: 'left' | 'right';
+}
+
+export const ScrollArea = React.memo(function ScrollArea(props: ScrollAreaProps) {
+  const { scrollDirection } = props;
+  const rootRef = React.useRef<HTMLDivElement>(null);
+  const api = React.useContext(ApiContext);
+  const rafRef = React.useRef(0);
+  const timeout = React.useRef<number>();
+  const [dragging, setDragging] = React.useState<boolean>(false);
+  const scrollPosition = React.useRef<ScrollParams>({
+    left: 0,
+    top: 0,
+  });
+
+  const handleScrolling = React.useCallback((newScrollPosition) => {
+    scrollPosition.current = newScrollPosition;
+  }, []);
+
+  const handleDragOver = React.useCallback(
+    (event) => {
+      let diff;
+
+      if (scrollDirection === 'left') {
+        diff = event.clientX - rootRef.current!.getBoundingClientRect().right;
+      } else if (scrollDirection === 'right') {
+        diff = Math.max(1, event.clientX - rootRef.current!.getBoundingClientRect().left);
+      } else {
+        throw new Error('wrong dir');
+      }
+
+      diff = (diff - 1) * 1.5 + 1;
```

Comment:
what are those numbers? is 1.5 arbitrary? What is it? it needs CONSTANT

Reply:
1 is the cliff, 1.5 is the slop.

Reply:
Ok will you remember in 2 years? What if it is someone else picking up the code?

Reply:
Agree, I recall trying to enforce no-magic number eslint once, I gave up 😆

Reply:
Done